src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.LeaseProviders
+{
+    /// <summary>
+    /// Aquired lease, which contains the resource key, duration of the lease and the token you can use to renew the lease with
+    /// </summary>
+    public class AcquiredLease
+    {
+        public string ResourceKey { get; }
+        public TimeSpan Duration { get; }
+        public string Token { get; }
+    }
+
+    /// <summary>
+    /// Lease request where you can specify ResourceKey, which you want to apply a lease on, and duration of your lease. 
+    /// </summary>
+    public class LeaseRequest
+    {
+        public string ResourceKey { get; set; }
+        public TimeSpan? Duration { get; set; }
+    }
+
+    /// <summary>
+    /// Lease provider interface 
+    /// </summary>
+    public interface ILeaseProvider
```

Comment:
But scalability is a implementation detail, which shouldn't be affecting API design, right? API design is expressing the caller side's demand. 
But I do agree that API should only contain the minimum set of functionality which are needed . Extra functionality needed on minor cases should come with extension on top of the base API. And I don't see query api is a must have for leaseProvider. So I think we reached the concensus to not include query API. 

BTW, I didn't include it in this PR since we were still discussing it. Just so you know that I addressed all the comments, and the PR is ready for a new round of review.